### ISSUE TITLE
Logits rescaling with temperature n times when non deterministic mode is applied

### ISSUE THREAD

**Issue by fhidalgor** [Adibvafa/CodonTransformer#19]
**Describe the bug**
When deterministic=False, num_sequences>1, the logits inside sample_non_deterministic() in CodonPrediction.py get scaled with temperature n times, where n is num_sequences.

**Where it happens**
logits /= temperature <----- this is making the logits be scaled in each iteration of the loop
probs = torch.softmax(logits, dim=-1)

**Possible solution**
Perform scaling insided softmax so logits dont get updated n times
  #logits /= temperature
  probs = torch.softmax(logits/temperature, dim=-1)

**My env**
Linux
Python 3.13.1

**Comment by Adibvafa** [comment]
Thank you for opening the issue, fixed.
